### ISSUE TITLE
CA1508 should not warn on as expression

### ISSUE THREAD

**Issue by JohanLarsson** [dotnet/roslyn-analyzers#4411]
```cs
class A
{
    void M(object o)
    {
        if (o is A a)
        {
        //  ↓↓↓↓↓ CA1508
            (a as B)?.ToString();
        }
    }
}

class B : A
{
}
```

![image](https://user-images.githubusercontent.com/1640096/98296556-60e29080-1fb3-11eb-811a-720761d9fc27.png)
